### PR TITLE
Upgrade to Elasticsearch 7.x

### PR DESCRIPTION
**Pull request by cmark**
This PR bumps the version of Elasticsearch to `7.5.2`.

Due to API and client changes in Elasticsearch it is necessary to make the following changes:
- [x] Remove the `scrollId` from the Java API (but still keep it in the low-level index client) 
- [x] ~Reduce the default initial number of shards to one in all repositories (from the default `6`)~ (won't do, keeping the default `6` instead of `1`, can be configured in `repository.index.numberOfShards` configuration setting)
- [x] Allow unlimited filter values in the Java API but use partitioned terms queries in Elasticsearch searches
- [x] Eliminate all warnings logged by Elasticsearch APIs

### CODE REVIEW

File: commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java [hunk 1]
```diff
@@ -190,7 +189,7 @@ public EsDocumentSearcher(EsIndexAdmin admin, ObjectMapper mapper) {
 			throw new IndexException("Couldn't execute query: " + e.getMessage(), null);
 		}
 		
-		final int totalHits = (int) response.getHits().getTotalHits();
+		final int totalHits = (int) response.getHits().getTotalHits().value;
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java]: Accuracy of total hits counting changed with ES 7.0 ("Track total hits up to 10,000 by default"): https://www.elastic.co/guide/en/elasticsearch/reference/current/release-notes-7.0.0.html

Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java]: Thanks! Fixed in commit 562a6e6.

File: commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java [hunk 2]
```diff
@@ -145,24 +145,23 @@ public void create() {
...
 					.putAll(toProperties(mapping))
-					.build());
+					.build();
 			
 			if (exists(mapping)) {
 				// update mapping if required
 				ImmutableOpenMap<String, MappingMetaData> currentIndexMapping;
 				try {
-					currentIndexMapping = client.indices().getMapping(new GetMappingsRequest().indices(index).types(type)).mappings().get(index);
+					currentIndexMapping = client.indices().getMapping(new GetMappingsRequest().types(type).indices(index)).mappings().get(index);
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java:159]: Is `types` still needed here?

Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java:159]: Yes, unfortunately, the GetMapping/PutMapping/CreateIndex action requests still require `type` value to be specified. The new ES HLRC however uses a different set of Request classes which do not require the `type` value. I was able to remove all other `type` fields from the `EsClient` interface and implementations.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/SearchResourceRequestIterator.java [hunk 3]
```diff
@@ -84,7 +72,7 @@ protected R computeNext() {
 		}
 
 		// Update scrollId and visited counter
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/SearchResourceRequestIterator.java]: This comment should be updated as well.

File: snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java [hunk 4]
```diff
@@ -87,7 +87,6 @@
 	private static final Logger LOG = LoggerFactory.getLogger("reasoner");
 
 	private static final int SCROLL_LIMIT = 10_000;
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java:89]: Now that `SearchResourceRequestIterator` relies on the revision system for snapshot semantics, limit constants like the above could be changed to eg. `BATCH_SIZE` or similar.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java [hunk 5]
```diff
@@ -251,6 +252,8 @@ void setLocales(List<ExtendedLocale> locales) {
 
 	@Override
 	public ExportResult execute(final RepositoryContext context) {
+		// register export start time for later use
+		final long exportStartTime = Instant.now().toEpochMilli();
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java:256]: How about using the export reference branch's head timestamp? We could even run the export on a `BranchContext` just to ensure that it's not a moving target.

Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java:256]: Great idea, but I'm afraid we cannot change the `RepositoryContext` to `BranchContext` in the `SnomedRF2ExportRequest` now. Feel free to experiment with it.

File: commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java [hunk 6]
```diff
@@ -556,9 +556,9 @@ private void bulkIndexByScroll(final EsClient client,
 				
 				final BulkByScrollResponse response; 
 				if ("update".equals(command)) {
-					response = client.updateByQuery(getTypeIndex(mapping), mapping.typeAsString(), BATCHS_SIZE, script, getConcurrencyLevel(), query);
+					response = client.updateByQuery(getTypeIndex(mapping), BATCHS_SIZE, script, getConcurrencyLevel(), query);
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java]: ⚠️ typo in constant

**Comment by apeteri** [comment]
🥇

**Review by apeteri — APPROVED** [review]

**Comment by cmark** [comment]
Thanks @apeteri!

**Pull request merged**